git/config.py

```diff
@@ -443,7 +452,49 @@ def string_decode(v):
             raise e
 
     def _has_includes(self):
-        return self._merge_includes and self.has_section('include')
+        return self._merge_includes and len(self._included_paths())
+
+    def _included_paths(self):
+        """Return all paths that must be included to configuration.
+        """
+        paths = []
+
+        for section in self.sections():
+            if section == "include":
+                paths += self.items(section)
+
+            match = CONDITIONAL_INCLUDE_REGEXP.search(section)
+            if match is not None and self._repo is not None:
+                keyword = match.group(1)
+                value = match.group(2).strip()
+
+                if keyword in ["gitdir", "gitdir/i"]:
+                    value = osp.expanduser(value)
+
+                    if not any(value.startswith(s) for s in ["./", "/"]):
+                        value = "**/" + value
+                    if value.endswith("/"):
+                        value += "**"
+
+                    # Ensure that glob is always case insensitive if required.
+                    if keyword.endswith("/i"):
+                        value = re.sub(
+                            r"[a-zA-Z]",
+                            lambda m: "[{}{}]".format(
+                                m.group().lower(),
+                                m.group().upper()
+                            ),
+                            value
+                        )
+
+                    if fnmatch.fnmatchcase(self._repo.git_dir, value):
+                        paths += self.items(section)
+
+                elif keyword == "onbranch":
+                    if fnmatch.fnmatchcase(self._repo.active_branch.name, value):
```

Comment:
What happens if there is no active branch? This could happen when `HEAD` is detached, i.e. points to a commit directly.
I would assume that `active_branch` is then `None` or throws, and this should be handled here. It's probably easy to test for once it's clear what actually happens then.

Reply:
True, it seems that a `TypeError` exception is raise when `HEAD` is detached
https://github.com/gitpython-developers/GitPython/blob/master/git/refs/symbolic.py#L272

```console
> repo = git.Repo("/somewhere")
> repo.active_branch
...
TypeError: HEAD is a detached symbolic reference as it points to 'ae451cd89c1b727ba632066c8b8a80ab8d22a0f8'
```

I'll had a `try.. except..` to just ignore this `includeIf` when branch cannot be retrieved.

Reply:
Added